src/cephadm/tox.ini

```diff
@@ -34,7 +34,7 @@ addopts =
 [testenv]
 skip_install=true
 deps =
-  pyfakefs == 5.0
+  pyfakefs >= 4
```

Comment:
just out of curiosity, do you know why the pyfakefs was pinned to 5.0? do we need any special features that are only available on that version?

Reply:
@rkachach that change was introduced in 52f5c2fed057efa25e1cbea2a2bb4295f77831e4 . i wanted to ensure that we are working with a known-to-work combination of dependencies instead of working in a flux environment. probably a better solution is to set the `python_version` as well? something like

```
pyfakefs == 4.5.6 ; python_version < 3.7;
pyfakefs == 5.0 ; python_version >= 3.6;
```

just checked on https://pypi.org/project/pyfakefs/4.5.6/ . it seems that 4.5.6 is the last version which supports python 3.6.

Reply:
Sounds mostly good. I checked up on [pep-508](https://peps.python.org/pep-0508/) because I have not used that distro selection syntax previously. It looks OK to me and I think that would accomplish the goal here of also working on py3.6 for centos 8.
What I'm not sure about is the use of `==` for the py3.6 and greater side. I also reviewed [pep-440](https://peps.python.org/pep-0440/) and it says:
```
The use of == (without at least the wildcard suffix) when defining dependencies for published distributions is strongly discouraged as it greatly complicates the deployment of security fixes. The strict version comparison operator is intended primarily for use when defining dependencies for repeatable deployments of applications while using a shared distribution index.
```
So, I think we'd at least want `pyfakefs >= 5`. I'm not sure why we'd want to exclude 5.y (5.1, 5.2, etc) releases. But I'm not sure what your intent with `pyfakefs == 5.0` is. Maybe you can clarify what your goal was and we can then come to a shared conclusion?  

I'm guessing at what you answer might be: if we think this is well-enough behaved project they'll keep API breaking changes only to major versions and so a rule like `>=5,<6` would be OK?

Reply:
@tchaikov any thoughts on the above comment? ^^^ thanks!